src/python/turicreate/toolkits/sound_classifier/_mx_sound_classifier.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2019 Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can
+# be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+
+
+import mxnet as mx
+import time
+import turicreate as _tc
+
+from .._mxnet import _mxnet_utils
+
+class MultiLayerPerceptronMXNetModel():
+
+
+	def __init__(self, feature_output_length, num_labels, custom_layer_sizes, verbose):
+
+		self.ctx = _mxnet_utils.get_mxnet_context()
+		self.verbose = verbose
+		self.custom_NN = self._build_custom_neural_network(feature_output_length, num_labels, custom_layer_sizes)
+		self.custom_NN.initialize(mx.init.Xavier(), ctx=self.ctx)
+
+		self.trainer = mx.gluon.Trainer(self.custom_NN.collect_params(), 'nag', {'learning_rate': 0.01, 'momentum': 0.9})
+		self.softmax_cross_entropy_loss = mx.gluon.loss.SoftmaxCrossEntropyLoss()
+
+
+	def train(self, data, label):
+		# Inside training scope
+		data_shape = data.shape[0]
```

Comment:
What is a typical value of `data_shape`? Isn't this really just the batch size?

Reply:
It can be <=`batch_size` depending on the size of data passed. If input data is less than batch size, `data_shape` would have a value less than batch size.